test/distributed/_sharded_tensor/test_sharded_tensor.py

```diff
@@ -799,3 +799,107 @@ def test_with_rpc_names(self):
                 self.assertEqual(rpc_rank, remote_shard.owner().id)
                 shard = remote_shard.to_here()
                 self.assertEqual((5, 5), shard.tensor.size())
+
+    @with_comms
+    @skip_if_lt_x_gpu(4)
+    @requires_nccl()
+    def test_init_from_local_shards(self):
```

Comment:
Can we add the following tests:

1. Some ranks don't have local shards.
2. Use a process group which is subset of default group using `new_group`.
3.  overlapping shards.
4. shards with gaps.